sh/gtf/validate_genome_references.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+from ftplib import FTP
+import argparse, re, os, sys
+
+fail = False
+def parse_url(url):
+    """
+    Parses a complete ftp URL into server, path and file name.
+    """
+    match = re.search("ftp://([a-z\.]*)/(.*)$", url)
+    server = match.group(1)
+    path_tokens = match.group(2).split("/")
+    return server, "/"+"/".join(path_tokens[:-1]), path_tokens[-1]
+
+def check_connection(organism, server, corrected_url, path, file):
+    ftp = FTP(server)
+    ftp.login()
```

Comment:
Retry mechanism for FTP failures?
```suggestion
from ftplib import error_temp
import time

def connect_with_retry(server, retries=3):
    for i in range(retries):
        try:
            ftp = FTP(server)
            ftp.login()
            return ftp
        except error_temp as e:
            if i < retries - 1:
                time.sleep(2)
            else:
                raise e

def check_connection(organism, server, corrected_url, path, file):
    ftp = connect_with_retry(server)
```

Reply:
This fails with

```
Run python sh/gtf/validate_genome_references.py --ensembl 114 --ensemblgenomes 61
Traceback (most recent call last):
  File "sh/gtf/validate_genome_references.py", line 127, in <module>
    check_connection(organism, server, corrected_fa, path, file)
  File "sh/gtf/validate_genome_references.py", line 38, in check_connection
    ftp.login()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/ftplib.py", line 408, in login
    resp = self.sendcmd('USER ' + user)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/ftplib.py", line 277, in sendcmd
    return self.getresp()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/ftplib.py", line 250, in getresp
    raise error_perm(resp)
ftplib.error_perm: 530 Can't change from guest user.
Error: Process completed with exit code 1.
```

Reply:
working now